sql/core/src/test/resources/sql-tests/results/join-lateral.sql.out

```diff
@@ -80,6 +80,46 @@ struct<c1:int,c2:int,c1:int,c2:int>
 1	2	0	3
 
 
+-- !query
+SELECT * FROM t1, LATERAL (SELECT t1.*)
+-- !query schema
+struct<c1:int,c2:int,c1:int,c2:int>
+-- !query output
+0	1	0	1
+1	2	1	2
+
+
+-- !query
+SELECT * FROM t1, LATERAL (SELECT t1.*, t2.* FROM t2)
+-- !query schema
+struct<c1:int,c2:int,c1:int,c2:int,c1:int,c2:int>
+-- !query output
+0	1	0	1	0	2
+0	1	0	1	0	3
+1	2	1	2	0	2
+1	2	1	2	0	3
+
+
+-- !query
+SELECT * FROM t1, LATERAL (SELECT t1.* FROM t2 AS t1)
+-- !query schema
+struct<c1:int,c2:int,c1:int,c2:int>
+-- !query output
+0	1	0	2
+0	1	0	3
+1	2	0	2
+1	2	0	3
+
+
+-- !query
+SELECT * FROM t1, LATERAL (SELECT t1.*, t2.* FROM t2, LATERAL (SELECT t1.*, t2.*, t3.* FROM t2 AS t3))
```

Comment:
Q: We cannot accept this query? It seems pogresql can though;
```
postgres=# SELECT * FROM t1, LATERAL (SELECT t1.*, t2.* FROM t2, LATERAL (SELECT t1.*, t2.*, t3.* FROM t2 AS t3) t) t;
 c1 | c2 | c1 | c2 | c1 | c2 
----+----+----+----+----+----
  0 |  1 |  0 |  1 |  0 |  2
  0 |  1 |  0 |  1 |  0 |  2
  0 |  1 |  0 |  1 |  0 |  3
  0 |  1 |  0 |  1 |  0 |  3
  1 |  2 |  1 |  2 |  0 |  2
  1 |  2 |  1 |  2 |  0 |  2
  1 |  2 |  1 |  2 |  0 |  3
  1 |  2 |  1 |  2 |  0 |  3
```

Reply:
Good question. Current Spark can only resolve the subquery using the immediate outer query. In this case t1.* can only be resolved using output from t2, which is c1, c2. There are many other places need to be updated to support nested subqueries with deep correlations.
https://github.com/apache/spark/blob/9c157a490bb59e02dcf44b14b411ea5beb68c238/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala#L164-L173

Reply:
Ah, ok. Thank you for the explanation. If there is a possibility to support it in future work, could you leave some comments there?

Reply:
Sounds good!